### ISSUE TITLE
Regarding the issue with the new version

### ISSUE THREAD

**Issue by manxswl** [ML4ITS/TimeVQVAE#14]
In the first stage of the new version, the reconstructed signal appears smoother compared to the old version. The GT and reconstructed signals are almost overlapping, but is this form correct?
![signal](https://github.com/user-attachments/assets/03ca6dae-985c-442b-8e69-3d4297e2fc85)

**Comment by danelee2601** [comment]
yes, the improvement is brought by using the Snake activation function instead of ReLU.

**Comment by danelee2601** [comment]
I guess it's often the case that we get hindsight about little simple things to tweak for further improvement.

**Comment by manxswl** [comment]
Your work's idea is very interesting, so I tried to reproduce your paper. In this process, I found that the reconstruction effects of the previous version and the later version differ greatly. I'm not sure which form is correct. The effect generated by the old version is shown in the image below. Is the activation function the only reason for this phenomenon?
![old_version](https://github.com/user-attachments/assets/d2b0d255-92db-4e78-a035-27e40630940a)

**Comment by manxswl** [comment]
And, I noticed that you have implemented some functionalities in both the encoder and decoder, and you've removed the loss updates related to zero padding. I don't quite understand why this was done.

**Comment by manxswl** [comment]
> Your work's idea is very interesting, so I tried to reproduce your paper. In this process, I found that the reconstruction effects of the previous version and the later version differ greatly. I'm not sure which form is correct. The effect generated by the old version is shown in the image below. Is the activation function the only reason for this phenomenon? ![old_version](https://private-user-images.githubusercontent.com/48723692/357313926-d2b0d255-92db-4e78-a035-27e40630940a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTQ5NjAsIm5iZiI6MTcyMzU1NDY2MCwicGF0aCI6Ii80ODcyMzY5Mi8zNTczMTM5MjYtZDJiMGQyNTUtOTJkYi00ZTc4LWEwMzUtMjdlNDA2MzA5NDBhLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDEzMTEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4MjNhMDBlOGIxMTEzNWM0YWM5OTI2NWY5MmNjMjdjZTkwNTIyMTEwN2IzZWM5MjVlZWNjMzkwNWFlOGU3M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OUeJdpxwyN11dw97fNJTleBALYimYkFLK888fQDu0v4)

I apologize, it was an issue with my dataset. This work is very interesting. Thank you for your explanations.

**Comment by manxswl** [comment]
> In the first stage of the new version, the reconstructed signal appears smoother compared to the old version. The GT and reconstructed signals are almost overlapping, but is this form correct? ![signal](https://private-user-images.githubusercontent.com/48723692/357274624-03ca6dae-985c-442b-8e69-3d4297e2fc85.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTQ5NzgsIm5iZiI6MTcyMzU1NDY3OCwicGF0aCI6Ii80ODcyMzY5Mi8zNTcyNzQ2MjQtMDNjYTZkYWUtOTg1Yy00NDJiLThlNjktM2Q0Mjk3ZTJmYzg1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDEzMTExOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiNGI3MWM2NTc5MDc0MDRkZTM1YWYxNWJlZTIwYWZhNjIwYjM1MGMxNTgyMzhjYWVkNzA1ZjIwZWM3MDY3NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qrcP-6_jw0naK9SEd-CVhaPgaMUGgoO7lSjGLH2nnOg)

The image after correctly introducing the dataset is shown below
![Uploading new_signal.jpg…]()

**Comment by manxswl** [comment]
After reproducing the results, I found that the reconstruction error for low-frequency signals is larger than before the update, and it's unstable.
![new_signal](https://github.com/user-attachments/assets/b89c975c-2869-46e5-b4ff-71a9e9e63759)

**Comment by danelee2601** [comment]
hmm.. I've noticed that the training on the FordA dataset generally resulted in unsatisfactory validation loss. Have you tried it on other datasets?

**Comment by danelee2601** [comment]
> And, I noticed that you have implemented some functionalities in both the encoder and decoder, and you've removed the loss updates related to zero padding. I don't quite understand why this was done.

I've simplified the code by leaving the essentials only so that end users can read and use it more easily.
In terms of performance, it's not heavily affected.

**Comment by manxswl** [comment]
I didn't use any other dataset. I just put all the instructions from the readme into a slurm script at once without paying attention, and I got the dataset wrong. I noticed that the log generated in the second stage is in this form (see image). Is this form correct? It seems a bit strange.
I just saw in the readme that the dataset has been changed to Wafer. I think I can try the effect again.
![media_images_visual comp (X_test vs Xhat)_8043_3c12001a4b29f3b360ad (1)](https://github.com/user-attachments/assets/e84f9797-1fac-4cca-945d-3a6985c230d8)
